Src/BootCamp.Chapter/ContactsCenter.cs

```diff
@@ -1,27 +1,66 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Text;
 
 namespace BootCamp.Chapter
 {
     public class ContactsCenter
     {
-        private readonly List<Person> _people;
+         private readonly string _peopleFile;
 
         public ContactsCenter(string peopleFile)
         {
-            // load people
+            if (!File.Exists(peopleFile))
+            {
+                throw new FileNotFoundException("File is made up"); 
+            }
+                      
+            _peopleFile = File.ReadAllText(peopleFile); 
+           
+            if (String.IsNullOrWhiteSpace(_peopleFile))
+            {
+                throw new Exception(); 
+            }
+         
         }
 
+       
         /// <summary>
         /// Gets people by filter predicate.
         /// </summary>
         /// <returns></returns>
         public List<Person> Filter(Predicate<Person> predicate)
         {
             var people = new List<Person>();
-            // ToDo: implement applying filter.
+            var splittedList = _peopleFile.Split($"{Environment.NewLine}");
+            for (int i = 1; i < splittedList.Length -1 ; i++)
+            {
+                var person = ConvertStringToPerson(splittedList[i]);
+                if (predicate(person))
+                {
+                    people.Add(person);
+                }
+
+            }
+                   
             return people;
         }
+
+        public Person ConvertStringToPerson(string personString)
```

Comment:
so doing this in a constructor is wrong ?   I just change the code to do this in a constructor

Reply:
Yes, it's bad to do this in the constructor because you couple the ctor to expect something that can be split always. Give more freedom of choice. However if you do want to parse, using a tryparse pattern is fine.

Reply:
oke, then will go back to the orginal idea and try to make a Tryparse. Have to think good how it must look like. Or I have to write one for string , one for dateTime and one for the Gender as Enum. Will sleep about that

Reply:
Do I need to use the TryParse or is this a sort of extra challenge.

I did now this : 
```
public Person ConvertStringToPerson(string personString)
        {
            var person = new Person(); 
            CultureInfo culture = new CultureInfo("en-US");
            var splittedPerson = personString.Split(',');
            person.Name = splittedPerson[0];
            person.SureName = splittedPerson[1];
            person.BirthDay = DateTime.Parse(splittedPerson[2], culture, DateTimeStyles.None);
            person.Gender = (Gender)Enum.Parse(typeof(Gender), splittedPerson[3]);
            person.Country = splittedPerson[4];
            person.Email = splittedPerson[5];
            person.StreetAdress = splittedPerson[6];
            
            return person; 
        }
```

Reply:
done and working